### DISCUSSION TITLE
current path in code?

### DISCUSSION THREAD

**Discussion by svhb1000** [zauberzeug/nicegui#2365] [Q&A]
Hello, 

Is there a way to interrogate the current URL/path when inside the code. 
As example: 
If you take the menu-generation in the website of nicegui itself, suppose you want to color the currently visible menu in a different color.  Is it possible to retrieve the current path, so you can apply a different styling for the item corresponding to the current page? 

Thanks.
Stefaan

**Comment by falkoschindler** [comment]
Here's one possibility:
```py
from nicegui import context

context.get_client().page.path
```

Or you ask the client:
```py
await ui.run_javascript('window.location.pathname')
```

Or you use FastAPI's `request` parameter:
```py
from fastapi import Request

@ui.page('/page')
def page(request: Request):
    ui.notify(request.url.path)
```

**Reply by Jonas0819** [reply]
Esto me sirvio, muchas gracias

---

Translation: "This helped me, thank you very much"

**Comment by agerwick** [comment]
ui.context.client.page.path

```python
from nicegui import ui
@ui.page('/page')
def page():
    ui.notify(ui.context.client.page.path)
```
